### ISSUE TITLE
Type-aware rule not detected in config file

### ISSUE THREAD

**Issue by chenglou** [oxc-project/oxc#15042]
`.oxlintrc.json`
```json
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "ignorePatterns": ["node_modules"],
  "plugins": [],
  "rules": {
    "typescript/await-thenable": "error",
    "no-debugger": "error"
  }
}
```

`test.ts`
```ts
await 12
debugger
```

Running `oxlint --type-aware test.ts` only logs the debugger but not the await error

**Comment by TheAlexLichter** [comment]
After thinking for a second longer about it: I think the behavior is correct as is but there should be a more clear response from the Oxlint side.

**Why**:

As no plugin is enabled (all are disabled via `plugins: []`, the typescript plugin is not on.
In turn, Oxlint can't tell where the rule should come from (e.g. from an internal plugin or a JS Plugin).

**How to handle**:

* Oxlint could emit an error here that a rule couldn't be found
* Or Oxlint could warn + try to resolve the rule "somehow"

**Bonus**:

* A way to enable single rules without the whole plugin, which is what you try at the moment.

> Once [a plugin is] enabled, rules in categories you have enabled will automatically be turned on.
